--- boost_1_55_0/boost/config/user.hpp	2014-11-25 16:35:44.355843130 -0800
+++ boost_1_55_0/boost/config/user.hpp	2014-11-25 16:52:46.299138270 -0800
@@ -21,6 +21,12 @@
 // https://code.google.com/p/android/issues/detail?id=42735#makechanges
 #define BOOST_ASIO_DISABLE_STD_ATOMIC 1
 
+// Fix for linux native:
+#define BOOST_ASIO_HAS_STD_ARRAY 1
+
+// Fix for android-{x86,mips}:
+#define BOOST_ASIO_DISABLE_SERIAL_PORT
+
 // define this to locate a compiler config file:
 // #define BOOST_COMPILER_CONFIG <myheader>
 
