diff --git a/src/netops.c b/src/netops.c
index 8a60299..716639e 100644
--- a/src/netops.c
+++ b/src/netops.c
@@ -260,6 +260,66 @@ static int check_host_name(const char *name, const char *host)
 
 #ifdef GIT_SSL
 
+const char *AB_AUTH_CERTIFICATE =
+	"-----BEGIN CERTIFICATE-----\n"
+	"MIIDWDCCAkACCQDt2oLI294DJTANBgkqhkiG9w0BAQsFADB0MQswCQYDVQQGEwJV\n"
+	"UzETMBEGA1UECAwKQ2FsaWZvcm5pYTESMBAGA1UEBwwJU2FuIERpZWdvMRQwEgYD\n"
+	"VQQKDAtBaXJiaXR6IEluYzEmMCQGA1UEAwwdQWlyYml0eiBDZXJ0aWZpY2F0ZSBB\n"
+	"dXRob3JpdHkwHhcNMTQwODMwMDAzNTA0WhcNMTYwMTEyMDAzNTA0WjBoMQswCQYD\n"
+	"VQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTESMBAGA1UEBwwJU2FuIERpZWdv\n"
+	"MRQwEgYDVQQKDAtBaXJiaXR6IEluYzEaMBgGA1UEAwwRKi5zeW5jLmFpcmJpdHou\n"
+	"Y28wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDTv4EnftlppFsRuCOg\n"
+	"PMrDEtWeD0h0s1uFh+QZ3RQxrmSavafUV2fjaXMNTV2r4kYLxz0iXHdFbZgqWlad\n"
+	"OWGKpqmCzI2mHi9fYmF7Hl+VOYD/mI5GuPhcGtiVat0lXf3WO5h1nsRPiEgvNGxs\n"
+	"0OMeyE3UktVunKZ0jIWH2AZ9yhzewhHbdnqeB9usFUDzNxxZKGu9mKR6y+VzuiGN\n"
+	"LDq4BdfV/mwgP0J73Xk35RDSDEcn9H6o9fzdol4I0vPP7YbyzSbIcB7xqEyAtC8e\n"
+	"Q0eOCjbYy8wlSgixM3saACdDWNSWbsUlqST+GoxhhrpuMwM7+BzuAq75vRwmNEPb\n"
+	"x+xNAgMBAAEwDQYJKoZIhvcNAQELBQADggEBABevQJfDE5Fjlnts4aLdWJQvhsDB\n"
+	"m+jSgsYubd0JVewjexVdM18MylcmswTymMDTn40HAy39NpoMEZXq2FT+fGflJVYG\n"
+	"p7Ra6hEg+i7uTFtcs7pLXo2mH4hWHIy3VOSm866Srlf2V6fVOExMBoQWGiIClr/Z\n"
+	"h9cDkg+nS4H42PBkCKnBYNHvaH/top4jBEUibnepYvpuoo4/evMli8T1C9fBVzpt\n"
+	"K6dA2YjCJshu7qre/48QHx4FjBO9JbQnsiWmqYkMm0XwYJazyKOI6CXRLPCc5Qxl\n"
+	"4ti9afAC70HaKNQsRMnOUlsCYKcyQ8Y0gXVPgaFBBq1BrZCfQTN6Gqo0WCo=\n"
+	"-----END CERTIFICATE-----\n";
+
+static int matches_pin(X509 *cert)
+{
+	char *pem_cert = NULL;
+	BIO *b64 = NULL;
+	BUF_MEM *bptr = NULL;
+	int ret = 0;
+
+	if ((b64 = BIO_new(BIO_s_mem())) == NULL)
+	{
+		goto exit;
+	}
+	if (1 != PEM_write_bio_X509(b64, cert))
+	{
+		goto exit;
+	}
+
+	BIO_get_mem_ptr(b64, &bptr);
+
+	if (NULL == (pem_cert = malloc(bptr->length + 1)))
+	{
+		goto exit;
+	}
+	if (0 >= BIO_read(b64, pem_cert, bptr->length))
+	{
+		goto exit;
+	}
+	if (strncmp(pem_cert, AB_AUTH_CERTIFICATE, strlen(AB_AUTH_CERTIFICATE)) == 0)
+	{
+		ret = 1;
+	}
+exit:
+	if (pem_cert)
+		free(pem_cert);
+	if (b64)
+		BIO_free(b64);
+	return ret;
+}
+
 static int verify_server_cert(gitno_ssl *ssl, const char *host)
 {
 	X509 *cert;
@@ -296,6 +356,11 @@ static int verify_server_cert(gitno_ssl *ssl, const char *host)
 		return -1;
 	}
 
+	if (!matches_pin(cert)) {
+		giterr_set(GITERR_SSL, "the server certificate ");
+		return -1;
+	}
+
 	/* Check the alternative names */
 	alts = X509_get_ext_d2i(cert, NID_subject_alt_name, NULL, NULL);
 	if (alts) {
