diff --git a/src/ecmult_impl.h b/src/ecmult_impl.h
index 80793db..15ba5ee 100644
--- a/src/ecmult_impl.h
+++ b/src/ecmult_impl.h
@@ -14,7 +14,7 @@
 
 // larger numbers may result in slightly better performance, at the cost of
 // exponentially larger precomputed tables. WINDOW_G == 14 results in 640 KiB.
-#define WINDOW_G 14
+#define WINDOW_G 12
 
 /** Fill a table 'pre' with precomputed odd multiples of a. W determines the size of the table.
  *  pre will contains the values [1*a,3*a,5*a,...,(2^(w-1)-1)*a], so it needs place for
