diff --git a/src/ecmult_impl.h b/src/ecmult_impl.h
index fd14bf12..340e5f43 100644
--- a/src/ecmult_impl.h
+++ b/src/ecmult_impl.h
@@ -35,10 +35,10 @@
     exponentially larger precomputed tables. */
 #ifdef USE_ENDOMORPHISM
 /** Two tables for window size 15: 1.375 MiB. */
-#define WINDOW_G 15
+#define WINDOW_G 11
 #else
 /** One table for window size 16: 1.375 MiB. */
-#define WINDOW_G 16
+#define WINDOW_G 12
 #endif
 #endif
