diff --git a/BRKey.c b/BRKey.c
index 9187c0f..0e989e6 100644
--- a/BRKey.c
+++ b/BRKey.c
@@ -52,8 +52,9 @@
 #ifndef __clang__
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
 #endif
-#include "secp256k1/src/basic-config.h"
-#include "secp256k1/src/secp256k1.c"
+#include "secp256k1_recovery.h"
+// #include "secp256k1/src/basic-config.h"
+// #include "secp256k1/src/secp256k1.c"
 #pragma clang diagnostic pop
 #pragma GCC diagnostic pop

diff --git a/BRKey.h b/BRKey.h
index fa0be09..174b810 100644
--- a/BRKey.h
+++ b/BRKey.h
@@ -27,6 +27,7 @@

 #include "BRInt.h"
 #include <stddef.h>
+#include <stdlib.h>
 #include <inttypes.h>

 #ifdef __cplusplus

diff --git a/ethereum/les/BREthereumLESBase.c b/ethereum/les/BREthereumLESBase.c
index 159fb21..3187b32 100644
--- a/ethereum/les/BREthereumLESBase.c
+++ b/ethereum/les/BREthereumLESBase.c
@@ -45,7 +45,8 @@

 #include "secp256k1.h"
 #include "secp256k1_ecdh.h"
-#include "secp256k1/src/hash.h"
+// #include "secp256k1/src/hash.h"
+#include "src/hash.h"

 #pragma clang diagnostic pop
 #pragma GCC diagnostic pop
